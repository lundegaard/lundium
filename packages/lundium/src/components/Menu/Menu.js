import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

const Menu = forwardRef(({ children, size, className, ...rest }, ref) => (
	<ul
		className={cx({ menu: true, [`menu--size-${size}`]: size }, className)}
		ref={ref}
		{...rest}
	>
		{children}
	</ul>
));

Menu.displayName = 'Menu';

Menu.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	size: PropTypes.oneOf(['sm']),
};

export default Menu;
