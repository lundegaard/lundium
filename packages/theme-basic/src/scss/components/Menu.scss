@import '../core';

/*
 -  Menu
 */


.menu {
	position: absolute;
	width: 100%;
	max-height: $menu-max-height;
	margin: logical ( -$input-border-width * 2 ) 0 0;
	padding: logical ( $input-border-width * 2 ) 0 0;
	list-style-type: none;
	border-block-start: none;
	background-color: $menu-bg;
	border-radius: 0 0 $border-radius $border-radius;
	box-shadow: $box-shadow-dropdown;
	color: $body-color;
	overflow-x: hidden;
	overflow-y: auto;
	z-index: $zindex-popover;

	&--no-gutters {
		margin: 0;
	}
}