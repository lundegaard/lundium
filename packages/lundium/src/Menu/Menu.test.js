import React from 'react';
import { shallow } from 'enzyme';

import Menu from './Menu';

describe('<Menu />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Menu {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
