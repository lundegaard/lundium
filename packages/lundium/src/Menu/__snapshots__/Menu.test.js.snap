// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> matches snapshot when shallowly rendered 1`] = `
<ul
  className="menu"
/>
`;
